File: Code Generation/responses/cpp/deepseek/easy/find-special-substring-of-length-k.cpp
#include <string>
using namespace std;

class Solution {
public:
    bool hasSpecialSubstring(string s, int k) {
        int n = s.length();
        
        for (int i = 0; i <= n - k; i++) {
            // Check if substring of length k starting at i has all same characters
            bool allSame = true;
            for (int j = i + 1; j < i + k; j++) {
                if (s[j] != s[i]) {
                    allSame = false;
                    break;
                }
            }
            
            if (allSame) {
                // Check character before substring (if exists)
                bool beforeValid = (i == 0) || (s[i - 1] != s[i]);
                
                // Check character after substring (if exists)
                bool afterValid = (i + k == n) || (s[i + k] != s[i]);
                
                if (beforeValid && afterValid) {
                    return true;
                }
            }
        }
        
        return false;
    }
};

File: Code Generation/responses/python/deepseek/easy/find-special-substring-of-length-k.py
class Solution:
    def hasSpecialSubstring(self, s: str, k: int) -> bool:
        n = len(s)
        
        for i in range(n - k + 1):
            # Check if substring of length k starting at i has all same characters
            substring = s[i:i+k]
            if len(set(substring)) != 1:
                continue
            
            # Check character before substring
            if i > 0 and s[i-1] == s[i]:
                continue
            
            # Check character after substring
            if i + k < n and s[i+k] == s[i]:
                continue
            
            # All conditions satisfied
            return True
        
        return False